### PR TITLE
add: License header in the top of the files in the smswithoutborders_libsig directory.

### PR DESCRIPTION
**Pull request by wuX4an**
Response of [this issue.](https://github.com/smswithoutborders/lib_signal_double_ratchet_python/issues/15)

### CODE REVIEW

File: smswithoutborders_libsig/helpers.py [hunk 1]
```diff
@@ -1,4 +1,8 @@
-
+"""
+This program is free software: you can redistribute it under the terms
+of the GNU General Public License, v. 3.0. If a copy of the GNU General
+Public License was not distributed with this file, see <https://www.gnu.org/licenses/>.
+"""
```
Comment (PromiseFru) [inline smswithoutborders_libsig/helpers.py:5]: Keep a single empty line after the top file docstring.

File: smswithoutborders_libsig/keystore.py [hunk 2]
```diff
@@ -1,4 +1,10 @@
-"""A module for managing a secure keystore."""
+"""
+This program is free software: you can redistribute it under the terms
+of the GNU General Public License, v. 3.0. If a copy of the GNU General
+Public License was not distributed with this file, see <https://www.gnu.org/licenses/>.
+"""
+
+# A module for managing a secure keystore.
```
Comment (PromiseFru) [inline smswithoutborders_libsig/keystore.py:7]: Keep the existing file description at the top of the docstring.

```suggestion
"""
A module for managing a secure keystore.
 
This program is free software: you can redistribute it under the terms
of the GNU General Public License, v. 3.0. If a copy of the GNU General
Public License was not distributed with this file, see <https://www.gnu.org/licenses/>.
"""

```

File: smswithoutborders_libsig/ratchets.py [hunk 3]
```diff
@@ -1,4 +1,10 @@
-"""A module that implements the ratchet mechanism."""
+"""
+This program is free software: you can redistribute it under the terms
+of the GNU General Public License, v. 3.0. If a copy of the GNU General
+Public License was not distributed with this file, see <https://www.gnu.org/licenses/>.
+"""
+
+# A module that implements the ratchet mechanism
```
Comment (PromiseFru) [inline smswithoutborders_libsig/ratchets.py:7]: Keep the existing file description at the top of the docstring.

**Review by PromiseFru — CHANGES_REQUESTED** [review]
Thank you, @wuX4an! Just a few more tweaks are needed. 😉

**Comment by wuX4an** [comment]
> > Response of [this issue.](https://github.com/smswithoutborders/lib_signal_double_ratchet_python/issues/15)
> 
> Tip
> 
> To link your pull request (PR) to the relevant issue and ensure that the issue is automatically closed once the PR is merged, you can use GitHub's supported keywords in the PR title.
> 
> For example, you can use keywords like "fixes," "closes," or "resolves" followed by the issue number. Here’s a format you can follow:
> 
> ```
> Fixes #123
> ```
> 
> This will automatically close issue `#123` when the PR is merged.
> 
> For more information on this, you can check the [GitHub documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue).

# Thanks for the tip.
I'm learning to contribute to projects and I'm only limited to markdowns and troubleshooting dependency errors in python.

**Comment by PromiseFru** [comment]
@wuX4an, please resolve the rest of the requested changes so I can merge the PR. 🙂

**Comment by wuX4an** [comment]
@PromiseFru I already did it, sorry, I didn't know xd.

**Comment by PromiseFru** [comment]
> @PromiseFru I already did it, sorry, I didn't know xd.

Hey @wuX4an, you marked the requests resolved without making the necessary file changes.
Please review the comments and update the files, ratchets.py and helpers.py, before marking the conversation as resolved. :)

https://github.com/smswithoutborders/lib_signal_double_ratchet_python/pull/16#discussion_r1775149792
https://github.com/smswithoutborders/lib_signal_double_ratchet_python/pull/16#discussion_r1775160032

**Comment by wuX4an** [comment]
> > @PromiseFru I already did it, sorry, I didn't know xd.
> 
> Hey @wuX4an, you marked the requests resolved without making the necessary file changes. Please review the comments and update the files, ratchets.py and helpers.py, before marking the conversation as resolved. :)
> 
> [#16 (comment)](https://github.com/smswithoutborders/lib_signal_double_ratchet_python/pull/16#discussion_r1775149792) [#16 (comment)](https://github.com/smswithoutborders/lib_signal_double_ratchet_python/pull/16#discussion_r1775160032)


Oh, F***k. I didn't realize xd

File: smswithoutborders_libsig/ratchets.py [hunk 4]
```diff
@@ -1,4 +1,10 @@
-"""A module that implements the ratchet mechanism."""
+"""
+A module for managing a secure keystore.
```
Comment (PromiseFru) [inline smswithoutborders_libsig/ratchets.py:2]:
```suggestion
A module that implements the ratchet mechanism.
```

**Review by PromiseFru — CHANGES_REQUESTED** [review]

**Review by PromiseFru — APPROVED** [review]
Thank you, @wuX4an!

**Pull request merged**